Declaring Java interface deprecation (#2365)
oneapi-src/oneDAL: README.md

- Use [Intel(R) Extension for Scikit-learn*](https://intel.github.io/scikit-learn-intelex/) to accelerate existing scikit-learn code without changing it
- Use [daal4py](https://github.com/intel/scikit-learn-intelex/tree/master/daal4py), a standalone package with Python API for oneDAL
- Use [Scala and Java interfaces](#apache-spark-mllib) that match Apache Spark MlLib API

> Deprecation Notice: Java interfaces in the oneDAL library have been deprecated and may no longer be supported in future releases.

Reviewer: ```suggestion
Deprecation Notice: The Java interfaces are deprecated in the oneDAL library and may no longer be supported in future releases.
```

<details><summary>Other Examples</summary>

- [MPI](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/cpp/mpi)
- [MySQL](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/cpp/mysql)

> Deprecation Notice: Java interfaces in the oneDAL library have been deprecated and may no longer be supported in future releases. This includes Scala, Spark and Hadoop samples

Reviewer: ```suggestion
Deprecation Notice: The Java interfaces are deprecated in the oneDAL library and may no longer be supported in future releases. This includes Scala, Spark, and Hadoop samples.
```
